### PR TITLE
Return StatusOr from createRouteSpecificFilterConfig for exception removal

### PR DESCRIPTION
**Pull request by mathetake**
Commit Message: Return StatusOr from createRouteSpecificFilterConfig for exception removal

Additional Description:
This makes createRouteSpecificFilterConfig method return the 
StatusOr wrapped pointer to prepare for removing exceptions
from extension factory code path. The number of changed files
is large but almost all change is just wrapping the return
type with absl::StatusOr.  

Risk Level: low
Testing: existing tests
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Comment by mathetake** [comment]
@alyssawilk thank you for the advice offline - I went ahead and opened the PR. I would appreciate it if you could take a look!

File: test/extensions/filters/http/bandwidth_limit/config_test.cc [hunk 1]
```diff
@@ -76,8 +79,11 @@ TEST(Factory, RouteSpecificFilterConfigDisabledByDefault) {
   NiceMock<Server::Configuration::MockServerFactoryContext> context;
 
   EXPECT_CALL(context.dispatcher_, createTimer_(_)).Times(0);
-  const auto route_config = factory.createRouteSpecificFilterConfig(
-      *proto_config, context, ProtobufMessage::getNullValidationVisitor());
+  const auto route_config =
```
Comment (alyssawilk) [inline test/extensions/filters/http/bandwidth_limit/config_test.cc:82]: If you do one of these PRs again I find a straight up pointer dereference to be a bit easier to read, but not worth another pass at CI.

**Review by alyssawilk — APPROVED** [review]

File: test/extensions/filters/http/bandwidth_limit/config_test.cc (hunk 1 above)
Comment (mathetake) [reply test/extensions/filters/http/bandwidth_limit/config_test.cc:82]: noted!

**Pull request merged**